src/Microsoft.DotNet.Build.Tasks/GenerateEncodingTable.cs

```diff
@@ -0,0 +1,395 @@
+﻿using Microsoft.Build.Framework;
```

Comment:
do you think it is good idea to regenerate the encoding table every time we build System.Text.Encoding.CodePages? I don't expect the encoding tables will not change that frequently and it doesn't make sense to have it build every time. 

I prefer to keep the current mechanism which the encoding data checked in corefx and have the tool to generate this file in some place in corefx (or other repo) and generate the data on demand. 

What you think?